### PR TITLE
9.1_Samil_ViewModel_1.3_OrderApp

### PR DESCRIPTION
**Pull request by smlblc**
9.1_Samil_Implement_ViewModel_1.3_OrderApp_Hw

### CODE REVIEW

File: 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var itemViewModel: ItemViewModel
+
+    private var whipedcream = false
+    private var chocolatechecked = false
+
+    lateinit var itemCount: TextView
```
Comment (Kryvovyaz) [inline 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/MainActivity.kt:21]: Mr. Samil,I think after you added   id 'kotlin-android-extensions' you can call view's items  directly by its name

File: 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        cream = findViewById(R.id.whipped_cream)
+        chocolate = findViewById(R.id.chocolate)
+        receipt = findViewById(R.id.reciept)
+        toppings = findViewById(R.id.toppings)
+        editText = findViewById(R.id.name_text)
+
+        itemViewModel = ViewModelProvider(this).get(ItemViewModel::class.java)
+        itemCount.text = itemViewModel.retrieveScore()
+
+        toppings.setOnClickListener { }
```
Comment (oliver-basoglu) [inline 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/MainActivity.kt:49]: What does it do? :)

File: 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var itemViewModel: ItemViewModel
+
+    private var whipedcream = false
+    private var chocolatechecked = false
```
Comment (oliver-basoglu) [inline 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/MainActivity.kt:19]: These 2 booleans should be in ViewModel

File: 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/ItemViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package com.example.a13_order_app_kotlin_viewmodel
+
+import androidx.lifecycle.ViewModel
+
+class ItemViewModel: ViewModel() {
+
+    private var item = 0
```
Comment (oliver-basoglu) [inline 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/ItemViewModel.kt:7]: count would be a better name

Comment (oliver-basoglu) [reply 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/ItemViewModel.kt:7]: or totalItem

File: 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/ItemViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        item += value
+    }
+
+    fun decreaseBy(value: Int) {
+        if (item > 0) {
+            item -= value
+        }
+    }
+
+    fun retrieveScore(): String {
```
Comment (oliver-basoglu) [inline 13_Order_App_Kotlin_ViewModel/app/src/main/java/com/example/a13_order_app_kotlin_viewmodel/ItemViewModel.kt:19]: retrieveItemCount()

**Pull request merged**